### PR TITLE
Add matching queue

### PR DESCRIPTION
**Pull request by anuanas2007**
This pull request introduces the implementation of the matching queue functionality in the backend service. It includes the setup of necessary connections, initialization of services, and handling of match requests.

### CODE REVIEW

File: backend/matching-service/src/index.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      const matchRequest = JSON.parse(request.content.toString());
+      const matchQueue = await getMatchQueue();
+      let matched = false;
+
+      // Matches with the same topic and difficulty
+      for (let i = 0; i < matchQueue.length; i++) {
+        if (
+          matchQueue[i].topic === matchRequest.topic &&
+          matchQueue[i].difficulty === matchRequest.difficulty &&
+          matchQueue[i].userId !== matchRequest.userId
```
Comment (freddychenyouren2) [inline backend/matching-service/src/index.js:40]: There are three conditions within this if-block. You may want to consider creating a boolean function that checks the matching criteria for better readability.

As an extension, you may want to consider having two types of such boolean functions:
1. `meetHardMatchingCriteria()`: Both `topic` and `difficulty` of the two users must equal before they are matched.
2. `meetSoftMatchingCriteria()`: Only `topic` of the two users are equal.

File: backend/matching-service/src/index.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          break;
+        }
+      }
+
+      // Matches with the same topic but different difficulty
+      if (!matched) {
+        for (let i = 0; i < matchQueue.length; i++) {
+          if (
+            matchQueue[i].topic === matchRequest.topic &&
+            matchQueue[i].userId !== matchRequest.userId
```
Comment (freddychenyouren2) [inline backend/matching-service/src/index.js:58]: Similar comment to above at line 40.

File: backend/matching-service/src/matchQueue.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    setTimeout(async () => {
+      const existingUser = await UserMatchRequest.findOne({
+        userId: userMatchRequest.userId,
+      });
+      if (existingUser) {
+        updateUser(userMatchRequest.userId, "timeout");
+        await UserMatchRequest.deleteOne({ userId: userMatchRequest.userId });
+      }
+    }, 60000);
```
Comment (freddychenyouren2) [inline backend/matching-service/src/matchQueue.js:50]: For the Timeout duration, 60 seconds may be too long. The Project Documentation and our plan in D1 are both 30 seconds. You may change the setTimeout duration from `60000` to `30000`.

**Review by freddychenyouren2 — CHANGES_REQUESTED** [review]
Some small changes and suggestions on code readability and time limit of matching service.

**Comment by btaanish** [comment]
Need the matching service backend to test out compatibity in the frontend side of things, will make the expected changes in my PR

**Pull request merged**